addons/sale/models/payment.py

```diff
@@ -146,18 +146,20 @@ def action_view_sales_orders(self):
     # Tools for payment
     # --------------------------------------------------
 
-    def render_sale_button(self, order, submit_txt=None, render_values=None):
+    def render_sale_button(self, order, submit_txt=None, render_values=None, amount=0):
         values = {
             'partner_id': order.partner_shipping_id.id or order.partner_invoice_id.id,
             'billing_partner_id': order.partner_invoice_id.id,
         }
         if render_values:
             values.update(render_values)
+        if not amount:
+            amount = order.amount_total
         # Not very elegant to do that here but no choice regarding the design.
         self._log_payment_transaction_sent()
         return self.acquirer_id.with_context(submit_class='btn btn-primary', submit_txt=submit_txt or _('Pay Now')).sudo().render(
             self.reference,
-            order.amount_total,
+            amount,
```

Comment:
you can directly do a ternary expression : 
`amount if amount else order.amount_total,`